var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -311,6 +313,14 @@ def edit(self, spec, prefix):
             os.environ["ESMF_NETCDF"] = "nc-config"
             os.environ["ESMF_NFCONFIG"] = "nf-config"
 
+            if spec["hdf5"].satisfies("~shared"):
```

Comment:
This shouldn't be needed either, `ESMF-NETCDF` and `ESMF_NFCONFIG` should be sufficient?